aleth-bootnode/main.cpp

```diff
@@ -0,0 +1,150 @@
+        createLoggingProgramOptions(c_lineWidth, loggingOptions));
+
+    po::options_description clientNetworking("NETWORKING", c_lineWidth);
+    auto addNetworkingOption = clientNetworking.add_options();
+#if ETH_MINIUPNPC
+    addNetworkingOption(
+        "upnp", po::value<string>()->value_name("<on/off>"), "Use UPnP for NAT (default: on)");
+#endif
+    addNetworkingOption("public-ip", po::value<string>()->value_name("<ip>"),
+        "Force advertised public IP to the given IP (default: auto)");
+    addNetworkingOption("listen-ip", po::value<string>()->value_name("<ip>(:<port>)"),
+        "Listen on the given IP for incoming connections (default: 0.0.0.0)");
+    addNetworkingOption("listen", po::value<unsigned short>()->value_name("<port>"),
+        "Listen on the given port for incoming connections (default: 30303)\n");
+
+    po::options_description allowedOptions("Allowed options");
+    allowedOptions.add(generalOptions).add(loggingProgramOptions).add(clientNetworking);
+
+    po::variables_map vm;
+    try
+    {
+        po::parsed_options parsed = po::parse_command_line(argc, argv, allowedOptions);
+        po::store(parsed, vm);
+        po::notify(vm);
+    }
+    catch (po::error const& e)
+    {
+        cout << e.what() << "\n";
+        return -1;
+    }
+
+    if (vm.count("help"))
+    {
+        cout << "NAME:\n"
+             << "   " << c_programName << "\n"
+             << "USAGE:\n"
+             << "   " << c_programName << " [options]\n\n";
+        cout << generalOptions << clientNetworking << loggingProgramOptions;
+        return 0;
+    }
+
+    /// Networking params.
+    string listenIP;
+    unsigned short listenPort = c_defaultIPPort;
+    string publicIP;
+    bool upnp = true;
+
+#if ETH_MINIUPNPC
+    if (vm.count("upnp"))
+    {
+        string m = vm["upnp"].as<string>();
+        if (isTrue(m))
+            upnp = true;
+        else if (isFalse(m))
+            upnp = false;
+        else
+        {
+            cerr << "Bad "
+                 << "--upnp"
+                 << " option: " << m << "\n";
+            return -1;
+        }
+    }
+#endif
+    if (vm.count("public-ip"))
+        publicIP = vm["public-ip"].as<string>();
+    if (vm.count("listen-ip"))
+        listenIP = vm["listen-ip"].as<string>();
+    if (vm.count("listen"))
+        listenPort = vm["listen"].as<unsigned short>();
+
+    setupLogging(loggingOptions);
+    if (loggingOptions.verbosity > 0)
+        cout << EthGrayBold << c_programName << ", a C++ Ethereum bootnode implementation" EthReset
+             << "\n";
+
+    auto netPrefs = publicIP.empty() ? NetworkConfig(listenIP, listenPort, upnp) :
+                                       NetworkConfig(publicIP, listenIP, listenPort, upnp);
+    auto netData = contents(getDataDir() / fs::path(c_networkConfigFileName));
+
+    // Pass in empty client version since it's not used in the discovery process
+    Host h("", netPrefs, &netData);
```

Comment:
Still I think better to give `c_programName` as client name; if someone tries to handshake with us, they'll see it.